test/utils/SubmissionFailureErrorCodesSpec.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import base.SpecBase
+import utils.SubmissionFailureErrorCodes._
+
+class SubmissionFailureErrorCodesSpec extends SpecBase {
+
+  "must return the correct SubmissionError" - {
+
+    Seq(
+      LocalReferenceNumberError,
+      ImportCustomsOfficeCodeError,
+      ExportCustomsOfficeNumberError,
+      ItemQuantityError,
+      InvalidOrMissingConsigneeError,
+      LinkIsPendingError,
+      LinkIsAlreadyUsedError,
+      LinkIsWithdrawnError,
+      LinkIsCancelledError,
+      LinkIsExpiredError,
+      LinkMissingOrInvalidError,
+      DirectDeliveryNotAllowedError,
+      ConsignorNotAuthorisedError,
+      RegisteredConsignorToRegisteredConsigneeError,
+      ConsigneeRoleInvalidError
+    ).foreach { submissionError =>
+
+      s"when given error code ${submissionError.code}" in {
+        SubmissionFailureErrorCodes.SubmissionError(SubmissionFailureErrorCodes.ConsigneeRoleInvalidError.code) mustBe ConsigneeRoleInvalidError
```

Comment:
I don't think `submissionError` is being used here.
```suggestion
        SubmissionError(submissionError.code) mustBe submissionError
```